Port ScriptBuffer to the new IPC serialization format

cdumez opened the pull request:
#### fe4d36309bff66d364e031740d1029b0fd4d692c
<pre>
Port ScriptBuffer to the new IPC serialization format
<a href="https://bugs.webkit.org/show_bug.cgi?id=268676">https://bugs.webkit.org/show_bug.cgi?id=268676</a>

Reviewed by Alex Christensen.

* Source/WebCore/workers/ScriptBuffer.cpp:
(WebCore::tryConvertToShareableResourceHandle):
(WebCore::ScriptBuffer::fromIPCData):
(WebCore::ScriptBuffer::ipcData const):
* Source/WebCore/workers/ScriptBuffer.h:
* Source/WebKit/Shared/WebCoreArgumentCoders.cpp:
(IPC::tryConvertToShareableResourceHandle): Deleted.
(IPC::ArgumentCoder&lt;WebCore::ScriptBuffer&gt;::encode): Deleted.
(IPC::ArgumentCoder&lt;WebCore::ScriptBuffer&gt;::decode): Deleted.
* Source/WebKit/Shared/WebCoreArgumentCoders.h:
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:

Canonical link: <a href="https://commits.webkit.org/274051@main">https://commits.webkit.org/274051@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9b936c24207dd50b2a33d68b6f92b04787b1bead

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/37736 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/16631 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/39978 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/40277 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/33576 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/39062 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/19277 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/13838 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/31935 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/38303 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/13987 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/33033 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/12230 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/12156 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/33720 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/41536 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/34125 "Passed tests") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/34155 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/38050 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/12756 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/10304 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/36224 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/14170 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/13140 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/4892 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/13481 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

achristensen07 commented on Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
```diff
@@ -5636,6 +5636,10 @@ struct WebCore::ServiceWorkerImportedScript {
 };
 
+[CreateUsing=fromIPCData] class WebCore::ScriptBuffer {
+    WebCore::ScriptBuffer::IPCData ipcData();
```
The metadata will need the definition of IPCData somewhere, too.  With similar things I've put std::variant here even though it has a more concise name.

cdumez replied:
Can I just add the typedef in the serialization.in? 
Using IPCData = …;

cdumez committed:
Port ScriptBuffer to the new IPC serialization format
https://bugs.webkit.org/show_bug.cgi?id=268676

Reviewed by Alex Christensen.

* Source/WebCore/workers/ScriptBuffer.cpp:
(WebCore::tryConvertToShareableResourceHandle):
(WebCore::ScriptBuffer::fromIPCData):
(WebCore::ScriptBuffer::ipcData const):
* Source/WebCore/workers/ScriptBuffer.h:
* Source/WebKit/Shared/WebCoreArgumentCoders.cpp:
(IPC::tryConvertToShareableResourceHandle): Deleted.
(IPC::ArgumentCoder<WebCore::ScriptBuffer>::encode): Deleted.
(IPC::ArgumentCoder<WebCore::ScriptBuffer>::decode): Deleted.
* Source/WebKit/Shared/WebCoreArgumentCoders.h:
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:

Canonical link: https://commits.webkit.org/274051@main
